refactor: Improve UX -style sidebar (add room, rooms & search)with bootstrap

dvshsahu opened the pull request:
change style sidebar of the app with bootstrap.

Made following changes

- style the search bar with bootstrap form-control class, files index.ejs
- style rooms section with cards, introduced 'card-columns' container to hold room details,
- each room has room name and description.
- move the add button at the bottom after the room cards, style with the card.
- refactor common code for appending room and content to the method and called from multiple places.

fixes: #103

Commit:
refactor: Improve UX - style sidebar (add room, rooms & search)with bootstrap

change style sidebar of the app with bootstrap.

Made following changes

- style the search bar with bootstrap form-control class, files index.ejs
- style rooms section with cards, introduced 'card-columns' container to hold room details,
- each room has room name and description.
- move the add button at the bottom after the room cards, style with the card.
- refactor common code for appending room and content to the method and called from multiple places.

fixes: #103

LalitNM commented:
There are few more things you should work upon.

- Style the leftbar-toggle button with bootstrap.

- There is a lot of unnecessary style in style.css which corresponds to `.left` or `. person` or some other classes used in leftbar, you need to remove this code from there because we don't want any unnecessary code there.

- You can also remove any custom styling of search bar at the top of leftbar(leave it as a bootstrap's default `form-control`.

Do these and then I will review your PR again.

dvshsahu commented:
Sure, I will be working on that, however need help on first point @LalitNM , 
- style the leftbar - toggle button with bootstrap : Here which button are we talking about, is it the add room button?

LalitNM commented:
No, the button referred here is the button which appears on small screens. check this screenshot.
![image](https://user-images.githubusercontent.com/63332774/95597184-d6ebea00-0a3d-11eb-83d9-74d3f5135e65.png)

dvshsahu commented:
okay sure

Commit:
refactor: remove unused css, style toggle button for small devices

    removed the unused css which included css for sidebar top and left section

    made following changes in the

    - remove unused css for left sidebar section
    - refactor css for general padding and font size, removed style and added bootstrap class for same
    - style toggle button css with bootstrap class for mobile device
    - added existing class person for toggle functionality.

    fixed: #103

LalitNM commented on public/style.css:
```diff
@@ -88,17 +90,15 @@ body {
+  width: 90%;
   height: 1px;
   content: '';
   background-color: #e6e6e6;
```
@dvshsahu can you think of something that we can get rid of this code also, perhaps a `container-fluid` can work(haven't tried it yet).

dvshsahu replied:
removed the code, now using plain <hr> tag to achieve the same

LalitNM reviewed (changes requested):
Everything else looks great except a small bug of content overflow on small screen. Screen-shot: 
![Screen Shot 2020-10-09 at 19 04 24](https://user-images.githubusercontent.com/63332774/95623598-18dc5680-0a65-11eb-84b5-af03c986b930.png)

Don't you think that we should put a scrollbar here?
And can you also increase the size of sidebar-btn?

dvshsahu commented:
Yes, scrollbar is needed there for both screens but I was under impression that separate issue will be there. No worries, I will take it up now.
Also, for button size, I feel this size is decent enough if the screen is too small, I will share screenshots with bigger sizes, we can discuss and fix this as well.

LalitNM commented:
@dvshsahu if you think that making a scrollbar is too much for one PR, then leave it. I will make another issue after merging this. And for the sidebar-btn I also think a little bit increase in font-size would be enough. But try to fix the changes in .top class. Then I will merge this.

dvshsahu commented:
I have already made some progress for scroll, I would like to complete it, is that fine @LalitNM 
For font size, sure thing I will go ahead and make the update.

Commit:
refactor : made the toggle button bigger, fix scroll issue for mobile and desktop, remove unused css

fix issue of sidebar rooms panel was not scrollable, made the toggle button a bit bigger  and removed the css for horizontal line after search bar

please go through the detailed changes
    - fix : sidebar was not scrollable if the more number of rooms are added, add room button was not visible for mobile and desktop devices, added overflow-style to achieve the same.
    - refactor : made the toggle button bigger for mobile device for improved UX, added btn-lg class to achieve the same.
    - refactor : removed the css that was used to provide horizontal line after the search section in sidebar, added hr tag with 0 margin botton ton achive the same.

fixes : #103

LalitNM committed:
update to  index.ejs

Removed `.top` form leftbar-top and added a class .`room-list`(styled in css in next commit) after removing `.overflow-style`.

LalitNM committed:
update to style.css

added room-list class and removed overflow-scroll property.
Else everything is just formatting of file.

LalitNM commented:
I merged:rocket: this @dvshsahu after some minor changes from my side. Hope you enjoyed working on this projest.

dvshsahu commented:
Thank you so much, it was a good learning experience. Hope to connect on this project again.
